File: Hahn Softwareentwicklung/Models/PaymentModel.cs
namespace Hahn_Softwareentwicklung.Models
{
    public class PaymentModel
    {
        public Guid Id { get; set; }
        public decimal Amount { get; set; }
        public PaymentMethodModel PaymentMethod { get; set; }
        public PaymentModel(Guid id, decimal amount, PaymentMethodModel paymentMethod) {
            Id = id;
            Amount = amount;
            PaymentMethod = paymentMethod;
        }
    }
}

File: Angular/Hahn_Softwareentwicklung/src/app/models/paymentModel.ts
import { PaymentMethodModel } from './paymentMethodModel';

export class PaymentModel {
    id: string;
    amount: number;
    paymentMethod: PaymentMethodModel;

    constructor(id: string, amount: number, paymentMethod: PaymentMethodModel) {
        this.id = id;
        this.amount = amount;
        this.paymentMethod = paymentMethod;
    }
}